File: WebApplication1/WebApplication1/ClientApp/src/app/models/article.ts
export class Article {
    id: string;
    title: string;
    content: string;
    views: number;
    rating: number;
    date: Date;
    difficulty: difficulties;
    language: languages;
}

export enum difficulties {
    Easy,
    Medium,
    Hard
};

export enum languages {
    CSharp,
    HTML_CSS,
    JS_TS
}

File: WebApplication1/WebApplication1/Models/Article.cs
using System;
using System.Reflection;

namespace WebApplication1.Models
{
    public class Article
    {
        public Guid ID { get; set; }

        public string Title { get; set; }

        public string Content { get; set; }

        public int Views { get; set; }

        public DateTime Date { get; set; }

        public Difficulties Difficulty { get; set; }

        public Languages Language { get; set; }

        public void Clone(Article model)
        {
            PropertyInfo[] properties = model.GetType().GetProperties();

            for (int i = 0; i < properties.Length; i++)
            {
                properties[i].SetValue(this, properties[i].GetValue(model));
            }
        }
    }
}
